src/rest/AnalyticsEndpoint.ts

```diff
@@ -93,22 +95,40 @@ export class AnalyticsEndpoint {
     return this.getFromService<string[]>(url, params);
   }
 
+  public forgetVisitorId() {
+    Cookie.erase('visitorId');
+  }
+
+  public kill() {
+    this.dead = true;
+  }
+
   private async sendToService(data: Record<string, any>, path: string, paramName: string): Promise<any> {
     // We use pendingRequest because we don't want to have 2 request to analytics at the same time.
     // Otherwise the cookie visitId won't be set correctly.
     if (AnalyticsEndpoint.pendingRequest != null) {
       await AnalyticsEndpoint.pendingRequest;
     }
+    if (this.dead) {
```

Comment:
why not add this check as the beginning of the method?

Reply:
I guess it's because you want to handle the case where you kill/disable the endpoint meanwhile a request is pending.
Although I'm not sure if it's absolutely necessary, it would still make sense to let what is ongoing finish and the next time sendToService() is called it wouldn't work. This would simplify the code IMO.

Reply:
You're right, I was worried that a request may still be ongoing and would create new local data.

This is resolved as I removed the ability to disable [AnalyticsEndpoint](../tree/master/src/rest/AnalyticsEndpoint.ts) directly.